source/basic.tex

```diff
@@ -833,7 +833,7 @@
 \pnum
 For a given declarative region \placeholder{R}
 and a point \placeholder{P} outside \placeholder{R},
-the set of \defnx{intervening}{region!declarative!intervening} declarative regions
+the set of \defna{intervening}{region!declarative} declarative regions
```

Comment:
Can this be `\defnadjx{intervening}{declarative region}{region!declarative}`? Similar question for various other uses of `\defna` below.

Reply:
Can you use `!` in the third argument to `\defnadjx` (_i.e._, in the `see{}` in `\indextext`)?

Reply:
I think we'd need some magic to turn `!`s into `, `s. I don't know if `\see` does that for us but it seems relatively straightforward to build.

Reply:
That said, should we really be indexing "declarative region" under "region" rather than under "declarative region" in the first place?

Reply:
We provide "a b, _see_ b, a" because it's standard to look up "noun, adjective", so maybe people do look for "region, declarative" even though the generic word "region" is not useful here?  (I also haven't checked to see whether the other uses of `\defna` are "misuses" like this.)

Reply:
Well, we don't have index entries for "binding, structured" or "operand, unevaluated". I think that's somewhat appropriate: these aren't "adjective noun" so much as an atomic noun phrase.

I think the right thing here is:

 * This should be `\defnadj{intervening}{declarative region}`
 * The "region, declarative" index entry should be changed to say "*see* declarative region" so that it points to this index sub-entry

Reply:
You mean `\defnadjx{intervening}{declarative regions}{declarative region}` for this case, right?  And are you saying that (`\defnadj` and) `\defnadjx` should be changed to _always_ index _adjective noun_ and generate a _see_ for _noun, adjective_?